Add new method `urlParameters` that takes a `Map`

brettwp opened the pull request:
Adding a new method `urlParameter(Map<String, Object> parameters)`.  I'm opening this as a draft PR to get some feedback.

brettwp committed:
Add failing tests

brettwp committed:
Move test func and add null check

brettwp committed:
Fix {@code}

robotdan commented on src/main/java/com/inversoft/rest/RESTClient.java:
```diff
@@ -351,6 +351,20 @@ public URI getURI() {
+   *                   this will call <code>urlParameter(String, Object)</code>
+   * @return This.
+   */
+  public RESTClient<RS, ERS> urlParameter(Map<String, Object> parameters) {
```
I would vote to name this `urlParameters` (plural)

voidmain replied:
Yeah. That's the Java convention, which is different than other languages. Multiples using `s` to be grammatically aligned.

brettwp replied:
Done.

robotdan reviewed (commented):
Looks good!

brettwp committed:
Bumb version, rename var, and move tests around with a better name

brettwp commented:
Last commit I bumped the version to `3.5.0`, changed `parameters` to `urlParameters`, and renamed the test method to better match the pattern `get_*`.  Also, changed draft PR to ready for review.

robotdan commented on src/main/java/com/inversoft/rest/RESTClient.java:
```diff
@@ -351,6 +351,20 @@ public URI getURI() {
+   *                      this will call <code>urlParameter(String, Object)</code>
+   * @return This.
+   */
+  public RESTClient<RS, ERS> urlParameter(Map<String, Object> urlParameters) {
```
My comment was actually to rename the method itself, but the parameter is also a good change. 

Should be :

```
public RESTClient<RS, ERS> urlParameters(Map<String, Object> urlParameters) {
   // Code
}
```

brettwp replied:
Oops.  I misunderstood what var needed changing.  I just pushed an update for this.

brettwp committed:
Fix method name

robotdan commented:
We can test with this PR - I think now that I approved it, you have the option to merge it to master.
